Helper/RuleConverter.php

```diff
@@ -0,0 +1,121 @@
+<?php
+/**
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade this module to newer
+ * versions in the future.
+ *
+ *
+ * @category  Smile
+ * @package   Smile\ElasticsuiteTargetRule
+ * @author    Richard BAYET <richard.bayet@smile.fr>
+ * @copyright 2017 Smile
+ * @license   Open Software License ("OSL") v. 3.0
+ */
+
+namespace Smile\ElasticsuiteTargetRule\Helper;
+
+use Magento\Framework\App\Helper\AbstractHelper;
+use Magento\Framework\App\Helper\Context;
+use Smile\ElasticsuiteCatalogRule\Model\RuleFactory;
```

Comment:
If the replacement happens in the Virtual Category module, this is independant of you will : 

- VirtualCategories module disabled : you use CatalogRule to build the queries, virtual categories are not handled properly (but they do not exist since VirtualCategories is disabled). You should test this case to ensure it is working properly.

- VirtualCategories module is enabled : you retrieve their factory and are able to handle virtual categories query properly. 

=> for me, there is no need to enforce dependency on VirtualCategories module, everything should work as explained above. If you enforce the dependency, one will not be able to disable VirtualCategories module without disabling yours.

Regards